### PR TITLE
Fix volume scheduling error handling

### PR DESCRIPTION
**Pull request by cofyc**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

fix incorrect error handling in PR https://github.com/kubernetes/kubernetes/pull/83343

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/controller/volume/scheduling/scheduler_binder.go [hunk 1]
```diff
@@ -523,7 +519,10 @@ func (b *volumeBinder) checkBindings(pod *v1.Pod, bindings []*bindingInfo, claim
 		}
 		selectedNode := pvc.Annotations[pvutil.AnnSelectedNode]
 		if selectedNode != pod.Spec.NodeName {
-			return false, fmt.Errorf("selectedNode annotation value %q not set to scheduled node %q", selectedNode, pod.Spec.NodeName)
```
Comment (cwdsuzhou) [inline pkg/controller/volume/scheduling/scheduler_binder.go]: IMO, the original error msg here may be better

Comment (cofyc) [reply pkg/controller/volume/scheduling/scheduler_binder.go]: See https://github.com/kubernetes/kubernetes/issues/83181 and discussion in pr #83343

Comment (cwdsuzhou) [reply pkg/controller/volume/scheduling/scheduler_binder.go]: Sorry, I see the issue but missed the pr.

Looks good. Users should not care the annotation. :)

**Comment by jsafrane** [comment]
/lgtm

Thanks for a quick fix!

**Pull request merged**